sqe-http-api/Services/ImagedObjectService.cs

```diff
@@ -71,46 +70,30 @@ public async Task<ImagedObjectListDTO> GetImagedObjectsAsync(uint? userId, uint
         }
         
         public async Task<ImagedObjectListDTO> GetImagedObjectsWithArtefactsAsync(uint? userId, uint editionId,
-            bool withMask = false)
+            bool withMasks = false)
         {
             var result = await GetImagedObjectsAsync(userId, editionId);
 
-            // Bronson: the entire Select and lambda expression should move to a static function ArtefactModelToDTO (probably in the ArtefactService)
-            var artefacts = (await _artefactRepository.GetEditionArtefactListAsync(userId, editionId, withMask)).Select(x => new ArtefactDTO()
-            {
-                id = x.artefact_id,
-                editionId = editionId,
-                mask = new PolygonDTO()
-                {
-                    mask = x.mask
-                },
-
-                // Bronson - I think we added a class that does this for you, and also parses the ID back into its components for querying
-                imagedObjectId = x.institution + "-" 
-                                                + x.catalog_number_1 
-                                                + (string.IsNullOrEmpty(x.catalog_number_2) ? "" : "-" + x.catalog_number_2),
-
-                name = x.name,
-                side = x.catalog_side == 0 ? ArtefactDTO.ArtefactSide.recto : ArtefactDTO.ArtefactSide.verso, 
-                zOrder = 0,
-                transformMatrix = "",
-            });
+            var artefacts = ArtefactDTOTransform.QueryArtefactListToArtefactListDTO(
+                (await _artefactRepository.GetEditionArtefactListAsync(userId, editionId, withMasks)).ToList(), 
+                editionId
+            );
 
             // Bronson: This is *really* unclear, GroupJoin is not something people usually understand.
             // A few comments are warranted here, or even better - dropping LINQ altogether here
             // I spent a couple of minutes reading GroupJoin's documenation, and I'm not closer to figuring out 
             // what is being grouped.
             result.imagedObjects = result.imagedObjects.GroupJoin(
```

Comment:
I'm still not clear on what this does. I think it's best to describe with a short example in the comments.

Reply:
I can do that.  I am also willing to drop Linq here.  I assume that this "functional" approach is optimized, but that may be an invalid assumption.
Basically I grab all the images requested (without artefact data), and I separately grab all the artefact data requested (I should have done that concurrently).  Then this code gathers the artefacts and inserts the proper ones into their corresponding image objects (so the artefacts in those image objects are no longer null).

Reply:
I think I prefer a comment, and keeping the functional approach. It'll force us (well, me, right now) to think more functionally. Don't worry about performance here, we will have zero CPU issues in the backend.

Reply:
Done.

Reply:
OK, but now there's a problem with the github interface - I don't see your new changes in this pull request...

Reply:
At the bottom of this page you see the commits that I have added since the review started.  I will continue making commits to this branch as I make the changes we have discussed here.  Then you just pull the request and everything will be done.